source/Calamari.AzureScripting/AzureContextScriptWrapper.cs

```diff
@@ -0,0 +1,135 @@
+﻿﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Calamari.Common.Features.EmbeddedResources;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Features.Scripting;
+using Calamari.Common.Features.Scripts;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+
+namespace Calamari.AzureScripting
+{
+    public class AzureContextScriptWrapper : IScriptWrapper
+    {
+        const string CertificateFileName = "azure_certificate.pfx";
+        const int PasswordSizeBytes = 20;
+        const string DefaultAzureEnvironment = "AzureCloud";
+
+        readonly ICalamariFileSystem fileSystem;
+        readonly ICalamariEmbeddedResources embeddedResources;
+        readonly IVariables variables;
+        readonly ScriptSyntax[] supportedScriptSyntax = {ScriptSyntax.PowerShell, ScriptSyntax.Bash};
+
+        public AzureContextScriptWrapper(IVariables variables, ICalamariFileSystem fileSystem, ICalamariEmbeddedResources embeddedResources)
+        {
+            this.variables = variables;
+            this.fileSystem = fileSystem;
+            this.embeddedResources = embeddedResources;
+        }
+
+        public int Priority => ScriptWrapperPriorities.CloudAuthenticationPriority;
+
+        public bool IsEnabled(ScriptSyntax syntax) => supportedScriptSyntax.Contains(syntax);
```

Comment:
The old code was checking for an account and !SF, does this mean we simply don't call this from SF?

```
   public bool IsEnabled(ScriptSyntax syntax) => variables.Get(SpecialVariables.Account.AccountType, "").StartsWith("Azure") &&
                                string.IsNullOrEmpty(variables.Get(SpecialVariables.Action.ServiceFabric.ConnectionEndpoint)) &&
                                supportedScriptSyntax.Contains(syntax);
```

Reply:
that is right @flin-8